src/Search.jsx

```diff
@@ -0,0 +1,136 @@
+import { useFetch } from "./hooks/useFetch";
+import { useContext, useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { ThemeContext } from "./ThemeContext";
+import { titleCase } from "./functions/titleCase";
+
+export const Search = () => {
+	const { theme } = useContext(ThemeContext);
+	const [searchInput, setSearchInput] = useState("");
+	const { data, loading, error } = useFetch(
+		"https://pokeapi.co/api/v2/pokedex/original-unova/"
+	);
+
+	const pokemon = data.pokemon_entries;
+
+	const searchBoxChange = (event) => {
+		const val = event.target.value.toLowerCase();
+		setSearchInput(val);
+	};
+
+	const displayResults = (list) => {
+		return list.map((entry) => {
+			return (
+				<div
+					className="box"
+					style={{
+						backgroundColor: theme.boxbg,
+						border: theme.border,
+						textAlign: "center",
+					}}
+					key={entry.id}
+				>
+					<Link
+						to={"../pokemon/" + entry.id}
+						style={{
+							color: theme.linkcolor,
+							padding: "1em",
+							fontSize: "1.25em",
+						}}
+					>
+						{titleCase(entry.name)}
+					</Link>
+				</div>
+			);
+		});
+	};
+
+	if (loading) {
+		return (
+			<img
+				src="https://media.tenor.com/On7kvXhzml4AAAAj/loading-gif.gif"
+				style={{ height: "50px" }}
+			/>
+		);
+		//if loading is true, then return the element and stop running the code
+	}
+
+	let simplifiedPokemonList = [];
+
+	pokemon.map((entry) => {
+		simplifiedPokemonList.push({
+			name: entry.pokemon_species.name,
+			id: entry.pokemon_species.url
```

Comment:
The `id` appears to be available from the API so you don't need to manipulate the string